### ISSUE TITLE
💅 `useImportExtensions` false positive with `.jsx` + colocated `.d.ts` file

### ISSUE THREAD

**Issue by chmmpagne** [biomejs/biome#8893]
### Environment information

<details>

```
CLI:
  Version:                      2.3.13
  Color support:                false

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v24.13.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.6.2

Biome Configuration:
  Status:                       Loaded successfully
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
```

</details>

### Rule name

useImportExtensions

### Reproduction repo

https://github.com/chmmpagne/biome-jsx-dts-repro

```bash
git clone https://github.com/chmmpagne/biome-jsx-dts-repro
cd biome-jsx-dts-repro
npx @biomejs/biome check src/Consumer.ts
```

Output:
```
src/Consumer.ts:1:23 lint/correctness/useImportExtensions  FIXABLE
  × Add a file extension for relative imports.

  > 1 │ import Component from './Component.jsx';

  i Safe fix: Add import extension .ts.
      import Component from "./Component.jsx.ts";
```

### Expected result

No diagnostic — importing `Component.jsx` is correct. The colocated `.d.ts` file provides type information only and should not cause a suggested extension change.

This is the same scenario fixed in #6829 (PR #6903) for `.js` + `.d.ts`, but the fix only matches `Some("js")`:

```rust
(Some("d"), Some("js")) if resolved_extension.is_some_and(|ext| ext == "ts") => {
    return None; // We resolved a `.d.ts` file, but imported the `.js` file: OK.
}
```

The `.jsx` + `.d.ts` variant is not handled because `existing_extension` is `"jsx"`, not `"js"`. The fix likely needs to also match `"jsx"` (and possibly `"mjs"` / `"cjs"` / `"mjsx"` for completeness).

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
Please provide a reproduction repo. You can also try multi file mode in the playground.

**Comment by chmmpagne** [comment]
Reproduction repo: https://github.com/chmmpagne/biome-jsx-dts-repro

```bash
git clone https://github.com/chmmpagne/biome-jsx-dts-repro
cd biome-jsx-dts-repro
npx @biomejs/biome check src/Consumer.ts
```

Output:
```
src/Consumer.ts:1:23 lint/correctness/useImportExtensions  FIXABLE
  × Add a file extension for relative imports.

  > 1 │ import Component from './Component.jsx';

  i Safe fix: Add import extension .ts.
      import Component from "./Component.jsx.ts";
```
